test/propagators.js

```diff
@@ -52,3 +54,91 @@ describe ('TextMapCodec', () => {
         assert.deepEqual(ctx.baggage, { 'some-key': 'some-value' });
     });
 });
+
+describe ('ZipkinB3TextMapCodec', () => {
+    it('correctly extract the zipkin headers from a span context', () => {
+        let codec = new ZipkinB3TextMapCodec({ urlEncoding: true });
+
+        let carrier = {
+            'x-b3-parentspanid': 'some-parent',
+            'x-b3-spanid': 'some-span',
+            'x-b3-traceid': 'some-trace',
+            'x-b3-sampled': '1',
+            'x-b3-flags': '1',
+            'foo': 'bar'
+        };
+
+        let ctx = codec.extract(carrier);
+        assert.equal(ctx.parentId, 'some-parent');
+        assert.equal(ctx.spanId, 'some-span');
+        assert.equal(ctx.traceId, 'some-trace');
```

Comment:
I think the codec should be returning a null or empty context because the values are not valid Zipkin IDs. Otherwise the thrift serialization will fail later anyway.

Reply:
Codec now returns an empty context, in the same manner as the `TextMapCodec`.